lit_llama/adapter_v2.py

```diff
@@ -0,0 +1,40 @@
+"""
+Utility functions to extend the original LLaMA-Adapter method to LLaMA-Adapter v2,
+based on the paper,
+LLaMA-Adapter V2: Parameter-Efficient Visual Instruction Model
+https://arxiv.org/abs/2304.15010
+"""
+import torch
+from torch import Tensor
+from torch.nn import functional as F
+
+from lit_llama.adapter import LLaMA
+
+
+def adapter_v2_mark_only_adapter_as_trainable(model: LLaMA) -> None:
+    """Sets `requires_grad=False` for all non-adapter weights."""
+    for name, param in model.named_parameters():
+        substrings = ("adapter_wte", "gating_factor", "adapter_scale", "adapter_bias", "RMSNorm")
+        if "RMSNorm" in name:
+            print(name)
```

Comment:
The named parameters are always of the form `module.submodule.subsubmodule....`. I think you confused it with the class name for RMSNorm. In order to mark the rms norm trainable, we would have to include

```
("rms", "ln_f") 
```
or we could be a bit more strict by explicitly naming

```
("rms_1", "rms_2", "ln_f") 
```

Reply:
(and there is a left-over debug print)

Reply:
Thanks, this should be addressed now. I refactored this into a function and broke it down for readability purposes in the future (which may be overkill). It's now

```python
def get_adapter_substrings():
    substrings = ["adapter_wte", "gating_factor"]  # regular adapter v1 parameters
    substrings.extend(["adapter_scale", "adapter_bias"])  # adapter v2: new bias and scale used in Linear
    substrings.extend(["rms_1", "rms_2", "ln_f"])  # adapter v2: RMSNorm parameters are now trainable
    return substrings
```